flang/lib/Lower/OpenMP/Utils.cpp

```diff
@@ -626,7 +634,16 @@ bool collectLoopRelatedInfo(
     collapseValue = evaluate::ToInt64(clause->v).value();
     found = true;
   }
+  std::int64_t sizesLengthValue = 0l;
+  if (auto *clause =
+          ClauseFinder::findUniqueClause<omp::clause::Sizes>(clauses)) {
+    sizesLengthValue = clause->v.size();
+    found = true;
+  }
 
+  collapseValue = collapseValue - sizesLengthValue;
+  collapseValue =
+      collapseValue < sizesLengthValue ? sizesLengthValue : collapseValue;
```

Comment:
This as well does not seem to affect on tests added/modified by the PR.

Let me know if I missed something.